aspnetcore/signalr/diagnostics.md

```diff
@@ -0,0 +1,206 @@
+| `Trace` | Very detailed diagnostic messages designed for diagnosing specific issues. |
+
+Once you've configured the verbosity, the logs will be written to the Browser Console (or Standard Output in a NodeJS application).
+
+If you want to send logs to a custom logging system, you can provide a JavaScript object implementing the `ILogger` interface. The only method that needs to be implemented is `log`, which takes the level of the event and the message associated with the event. For example:
+
+[!code-typescript[Creating a custom logger](diagnostics/custom-logger.ts?highlight=3-7,13)]
+
+## .NET client logging
+
+> [!NOTE]
+> Client-side logs may contain sensitive information from your application. **Never** post raw logs from production applications to public forums like GitHub.
+
+To get logs from the .NET client, you can use the `.ConfigureLogging` method on `HubConnectionBuilder`. This works the same way as the `.ConfigureLogging` method on `WebHostBuilder` and `HostBuilder`. You can configure the same logging providers you use in ASP.NET Core. However, you have to manually install and enable the NuGet packages for the individual logging providers.
+
+### Console logging
+
+In order to enable Console logging, add the `Microsoft.Extensions.Logging.Console` nuget package. Then, use the `AddConsole` method to configure the console logger:
+
+[!code-csharp[Configuring console logging in .NET client](diagnostics/net-client-console-log.cs?highlight=6)]
+
+### Debug output window logging
+
+You can also configure logs to go to the Output Window in Visual Studio. Install the `Microsoft.Extensions.Logging.Debug` package and use the `AddDebug` method:
+
+[!code-csharp[Configuring debug output window logging in .NET client](diagnostics/net-client-debug-log.cs?highlight=6)]
+
+### Other logging providers
+
+You can also use other logging providers such as Serilog, Seq, NLog, or any other logging system that integrates with `Microsoft.Extensions.Logging`. If your logging system provides an `ILoggerProvider`, you can register it with `.AddProvider`:
+
+[!code-csharp[Configuring a custom logging provider in .NET client](diagnostics/net-client-custom-log.cs?highlight=6)]
+
+### Controlling verbosity
+
+If the debug logs are too verbose, you can set the default minimum level to `Information` and just increase the logging level for SignalR logs. This can be done in code, in much the same was as on the server:
+
+[!code-csharp[Controlling verbosity in .NET client](diagnostics/logging-config-client-code.cs?highlight=8-10)]
+
+## Network traces
+
+> [!NOTE]
+> A network trace contains the full contents of every message sent by your application. **Never** post raw network traces from production applications to public forums like GitHub.
+
+If you encounter an issue, a network trace can sometimes provide a lot of helpful information. This is particularly useful if you are going to file an issue on our issue tracker.
+
+### Using Fiddler (preferred option)
```

Comment:
Although logically this is under Network traces and should be h3, that's a lot of content excluded from the right nav and forces some sections down to h4.  I suggest continuing with h2 for each mode of getting network traces.

Reply:
Ok, I think I can bump the h3s and h4s up.